src/core/plugins/core/plugin.js

```diff
@@ -203,19 +203,21 @@ class CorePlugin extends Plugin {
             }, 10);
           })
             .then(downloadedFilePath => {
-              fs.ensureDir(
-                path.join(this.cachePath, this.props.config.codename, group)
-              ).then(() =>
-                fs.copyFile(
-                  downloadedFilePath,
-                  path.join(
-                    this.cachePath,
-                    this.props.config.codename,
-                    group,
-                    file.name
-                  )
+              return fs
+                .ensureDir(
+                  path.join(this.cachePath, this.props.config.codename, group)
                 )
-              );
+                .then(() =>
+                  fs.copyFile(
+                    downloadedFilePath,
+                    path.join(
+                      this.cachePath,
+                      this.props.config.codename,
+                      group,
+                      file.name
+                    )
+                  )
+                );
             })
             .then(() =>
               checkFile(
```

Comment:
I wonder if this should return just like the invocation above? Otherwise `.then(ok` below gets undefined if I'm not mistaken. According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#return_value:

> If a handler function:
> ...
> - ...
> - doesn't return anything, the promise returned by then gets resolved with an undefined value.

Reply:
Never mind, you actually have a return here!

Just as I was writing about the syntax error you were observing - because this is `() => foo()`, and `() => return foo()` is invalid - it hit me. I was about to say that `return` has to be in a block, so `() => { return foo(); }`, but realized that `=>` without block already returns the value of the expression! That is why it works (otherwise the `if` below would have thrown because `undefined` is falsy).

In that way you can actually remove the `return` above as it's a single expression now, without the `{}` it'll return the value of the `fs.ensureDir().then...` chain.

(I'm Rusty :crab: on JS :wink:)

Reply:
As we finally figured out in our discussion, removing the curly braces accomplishes the same as adding a return inside them, and it fits more neatly with the rest of the surrounding code.